### ISSUE TITLE
False positive with addition of integer operands

### ISSUE THREAD

**Issue by marcospassos** [phpstan/phpstan-strict-rules#46]
The check `checkAlwaysTrueCheckTypeFunctionCall` incorrectly assumes that the result of adding two integers is always an integer:

```php
function addExact(int $augend, int $addend) : int
{
    $result = $augend + $addend;

    // Call to function is_int() with int will always evaluate to true.  
    if (!\is_int($result)) {
        throw new OverflowException('The result overflows the range of an integer.');
    }

    return $result;
}

addExact(\PHP_INT_MAX, 1);
```
